### DISCUSSION TITLE
Setting Initial value to decision variables in matrix form

### DISCUSSION THREAD

**Discussion by sarathsurpur** [coin-or/pulp#368] [Q&A] [answered]
how to initialise values to lpvariables in matrix form?
-----------------------

With reference to test5.py, I tried to solve a problem following the example specified in test5.py, in my case I have to initialise the value to LPvariable.matrix but when I try to do that I am getting an error

I tried the below piece of code, thanks in advance

dec_var = LpVariable.matrix("dec_var",(range(10),range(10)),0,1,cat = 'Binary')

for i in range(10):
    for j in range(10):
        dec_var[i][j].setInitialValue(temp_matrix[i][j])

**Comment by pchtsp** [comment, marked as answer]
Hello!

1. I have no idea what `test5.py` mean.
2. Could you please say what error you see.
3. Can you provide temp_matrix ?
4. Next time you double post in Stack Overflow and here, say so.
5. What version of pulp are you using?

This works for me:
```python
from pulp import *
dec_var = LpVariable.matrix("dec_var",(range(10),range(10)),0,1,cat = 'Binary')

for i in range(10):
    for j in range(10):
        dec_var[i][j].setInitialValue(1)
```

**Reply by sarathsurpur** [reply]
Thanks for the reply franco, I was able to run this. The issue was using older older version of pulp.

After updating to latest version I didn't face any issue, thanks for taking time to reply on this.
